### PR TITLE
clean getDescription

### PR DESCRIPTION
**Pull request by AntonPuko**

### CODE REVIEW

File: src/clients/Device.js [hunk 1]
```diff
@@ -327,14 +318,18 @@ class Device extends EventEmitter {
       });*/
 
       this._sendHello();
-
       this._connectionStartTime = new Date();
 
-      // We don't need to await this as other functions will await the response
```
Comment (AntonPuko) [inline src/clients/Device.js:333]: yes we do. function that await for this promise is `get attributes from database and setting state to READY`, if we don't await description, we can't guarantee strict status order. It should be strongly INTIAL -> GOT_HELLO -> GOT_DESCRIPTION -> READY.

File: src/clients/Device.js [hunk 2]
```diff
@@ -327,14 +318,18 @@ class Device extends EventEmitter {
...
-      this._getDescription();
+      const { functionState, systemInformation } = await this._getDescription();
```
Comment (AntonPuko) [inline src/clients/Device.js:323]: I return functionState and systemInformation from the _getDescription promise. I strongly believe that the more void functions we have, that actually should return something(but instead they assign something to private pros inside) the less readable and unpredictable our code is.

File: src/server/DeviceServer.js [hunk 3]
```diff
@@ -295,48 +295,45 @@ class DeviceServer {
...
-          device.setStatus(DEVICE_STATUS_MAP.GOT_REPO_ATTRIBUTES);
-
-          this.publishSpecialEvent(
-            SYSTEM_EVENT_NAMES.SPARK_STATUS,
-            'online',
-            deviceID,
+          const {
+            claimCode,
+            currentBuildTarget,
+            imei,
```
Comment (AntonPuko) [inline src/server/DeviceServer.js:307]: from where we actually get info like imei, isCellural, last_iccid ? Seems now its just always undefined.

Comment (jlkalberer) [reply src/server/DeviceServer.js:307]: From an electron -- I'll be testing one this weekend.

**Review by jlkalberer — APPROVED** [review]

**Pull request merged**